src/app/item/item-list/item-list.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component } from '@angular/core';
+import { GroceryService } from 'src/app/grocerservice.service';
+//import { Item } from '../models/item';
+import { Item } from 'src/app/models/item';
+
+@Component({
+  selector: 'app-item-list',
+  templateUrl: './item-list.component.html',
+  styleUrls: ['./item-list.component.css'],
+})
+export class ItemListComponent {
+  items: Promise<Item[]> = this.groceryService.items;
+  filteredItems: Promise<Item[]> = this.groceryService.items;
+  types: string[] = ['vegetable', 'fruit', 'all'];
+  index: number = 0;
+
+  constructor(private readonly groceryService: GroceryService) {
+    this.items = this.groceryService.items;
+    console.log(this.items);
+    this.filteredItems = this.groceryService.items;
+    console.log(this.items);
+  }
+
+  addToCart(item: Item) {
+    this.groceryService.addToCart(item);
+  }
+
+  filterType(value: object) {
+    if (this.types[this.index] == 'all') {
+      this.filteredItems = this.items;
+    } else {
+      this.filteredItems = this.items.then((x) =>
+        x.filter((a) => {
+          return a.type == this.types[this.index - 1];
+        })
+      );
+    }
+
+    this.index++;
+    if (this.index > 2) {
+      this.index = 0;
+    }
+  }
+
```

Comment:
Your sortPrice and sortName methods modify the original array, which might not be intentional. While sorting, you should create a new array to prevent side effects. This is especially important if this.items is used elsewhere or its original order is needed. You can use the spread operator or Array.from to clone the array before sorting